test/e2e/operator_test.go

```diff
@@ -1096,6 +1096,89 @@ func TestInternalLoadBalancerGlobalAccessGCP(t *testing.T) {
 	}
 }
 
+func TestAWSLBTypeChange(t *testing.T) {
+	t.Parallel()
+	platform := infraConfig.Status.Platform
+
+	if platform != "AWS" {
+		t.Skipf("test skipped on platform %q", platform)
+	}
+
+	name := types.NamespacedName{Namespace: operatorNamespace, Name: "awslb"}
+	ic := newLoadBalancerController(name, name.Name+"."+dnsConfig.Spec.BaseDomain)
+	ic.Spec.EndpointPublishingStrategy.LoadBalancer = &operatorv1.LoadBalancerStrategy{
+		Scope: operatorv1.ExternalLoadBalancer,
+	}
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller: %v", err)
+	}
+	defer assertIngressControllerDeleted(t, kclient, ic)
+
+	// Wait for the load balancer and DNS to be ready.
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, name, availableConditionsForIngressControllerWithLoadBalancer...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	lbService := &corev1.Service{}
+	if err := kclient.Get(context.TODO(), controller.LoadBalancerServiceName(ic), lbService); err != nil {
+		t.Fatalf("failed to get LoadBalancer service: %v", err)
+	}
+
+	if err := kclient.Get(context.TODO(), name, ic); err != nil {
+		t.Fatalf("failed to get ingresscontroller %s: %v", name, err)
+	}
+	if ic.Annotations == nil {
+		ic.Annotations = map[string]string{}
+	}
+
+	// Annotate the ingresscontroller to tell the operator to automatically
+	// delete and recreate the LoadBalancer service if necessary when the
+	// LB changes, and change the LB.
+
+	//Update to NLB
+	lb := &operatorv1.LoadBalancerStrategy{
+		Scope: operatorv1.ExternalLoadBalancer,
+		ProviderParameters: &operatorv1.ProviderLoadBalancerParameters{
+			Type: operatorv1.AWSLoadBalancerProvider,
+			AWS: &operatorv1.AWSLoadBalancerParameters{
+				Type: operatorv1.AWSNetworkLoadBalancer,
+			},
+		},
+	}
+	ic.Spec.EndpointPublishingStrategy.LoadBalancer = lb
+	if err := kclient.Update(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to update ingresscontroller: %v", err)
+	}
+
+	// Wait for the load balancer and DNS to be ready.
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, name, availableConditionsForIngressControllerWithLoadBalancer...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	err := wait.PollImmediate(5*time.Second, 5*time.Minute, func() (bool, error) {
+		service := &corev1.Service{}
+		if err := kclient.Get(context.TODO(), controller.LoadBalancerServiceName(ic), service); err != nil {
+			if apierrors.IsNotFound(err) {
+				return false, nil
+			}
```

Comment:
just added a check. do you want to delete it ? I think it does not harm to keep a check.

Reply:
It seems to me like this condition should never occur.  Anyway, either the condition cannot occur or it can.  If it cannot occur, then the check is pointless.  If it can but shouldn't occur, then why wouldn't you want the test to log it?

Reply:
OK will remove it.